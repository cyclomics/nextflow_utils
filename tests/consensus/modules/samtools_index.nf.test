nextflow_process {

    name "Test Process SamtoolsIndexWithID"
    script "consensus/modules/samtools.nf"
    process "SamtoolsIndexWithID"
    
    
    test("Samtools BAM indexing") {

        when {
            process {
                """
                sample_id = "SOME_ID_1"
                file_id = "SOME_ID_2"
                bam_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.bam")

                input[0] = Channel.of([sample_id, file_id, bam_file])
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
            
            def expected_bai = "FAW79986_pass_dbcafcd9_40623079_0_filtered.bam.bai"
            with(process.out[0]) {
                def result_bai = path(get(0)[3]).getFileName().toString()
                assert result_bai == expected_bai
            }
        }
    }
}
