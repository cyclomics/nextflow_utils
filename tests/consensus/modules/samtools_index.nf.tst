nextflow_process {

    name "Test Process SamtoolsIndexWithID"
    script "consensus/modules/samtools.nf"
    process "SamtoolsIndexWithID"
    
    
    test("Samtools BAM indexing") {

        when {
            process {
                """
                sample_id = "test_sample"
                bam_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.bam")

                input[0] = Channel.of([sample_id, bam_file])
                """
            }
        }
        then {
            assert process.success
            
            def bai_filename = "FAW79986_pass_dbcafcd9_40623079_0_filtered.bam.bai"
            with(process.out[0]) {
                assert path(get(0)[2]).getFileName().toString() == bai_filename
                assert path(get(0)[2]).size() > 0
            }
        }
    }
}
