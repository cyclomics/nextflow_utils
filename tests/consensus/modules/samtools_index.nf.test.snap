{
    "Samtools BAM indexing": {
        "content": [
            {
                "0": [
                    [
                        "SOME_ID_1",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.bam:md5,52245f75cbcf1e143cb005bbbab8776d",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.bam.bai:md5,14debf7a36a39f9de48e2cddbd5b5b1e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-02-05T15:23:49.439919"
    }
}